### DISCUSSION TITLE
Lesson-4 Fund function of fund me contract dosen't work on remix vm...why???

### DISCUSSION THREAD

**Discussion by Raghav-Programmer** [Cyfrin/foundry-full-course-cu#358] [Q&A]
In lesson-4 I tried to deploy the contract and it deployed fine......then I added some funds to value but fund function didn't work....is it because aggregatorv3interface has data feed address for sepolia testnet????

**Comment by ChefAharoni** [comment]
Hi @Raghav-Programmer, please follow the question formatting [guidelines](https://github.com/Cyfrin/foundry-full-course-f23/blob/main/how-to-ask-a-question.md).

Can you provide more data to your question?
Copy/paste some of your code and try be more specific about what you're trying to accomplish and what's wrong.

That way, it'll be much easier for everyone to help you :)

**Comment by Raghav-Programmer** [comment]
here is the code:
`// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {PriceConverter} from "./PriceConverter.sol";

contract FundMe {
    using PriceConverter for uint256;

    uint256 constant MIN_USD = 5e18;
    address[] funders;
    mapping(address funder => uint256 amountFunded) addressToAmountFunded; 

    address immutable i_owner;
    constructor() {
        i_owner = msg.sender;
    }

    function fund() public payable {
        require(msg.value.getConversionRate() >= MIN_USD, "Didn't Send Enough!!!");
        funders.push(msg.sender);
        addressToAmountFunded[msg.sender] += msg.value; 
    }

    function withdraw() public onlyOwner {
        for(uint256 funderIndex = 0; funderIndex < funders.length; funderIndex++) {
            addressToAmountFunded[funders[funderIndex]] = 0;
        }
        funders = new address[](0);

        //transfer
        //payable(msg.sender).transfer(address(this).balance);
        //send
        //bool sent = payable(msg.sender).send(address(this).balance);
        //require(sent, "Send Failed");
        //call
        (bool callSuccess,) = payable(msg.sender).call{value: address(this).balance}("");
        require(callSuccess, "Call Failed");
    }

    modifier onlyOwner() {
        require(msg.sender == i_owner, "Sender is Not Owner!!");
        _;
    }

    receive() external payable {
        fund();
    }

    fallback() external payable {
        fund();
    }

}`

**Comment by Raghav-Programmer** [comment]
I think it is easier to understand the question now
